### PR TITLE
DAOS-7944 test: Added more label support in TestPool methods

### PR DESCRIPTION
**Pull request by shimizukko**
Added label support in:
set_property() - dmg.pool_set_prop()
evict() - dmg.pool_evict()
exclude() - dmg.pool_exclude()
set_query_data() - dmg.pool_query()
reintegrate() - dmg.pool_reintegrate()
drain() - dmg.pool_drain()

Skip-unit-tests: true
Test-tag: pr test_simple_rebuild
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/tests/ftest/pool/evict_test.py [hunk 1]
```diff
@@ -252,9 +253,9 @@ def test_evict_bad_uuid(self):
         """
         Test evicting a pool using an invalid uuid.
 
-        :avocado: tags=all,pool,pr,daily_regression,full_regression,small
-        :avocado: tags=poolevict
-        :avocado: tags=pool_evict_bad_uuid,DAOS_5610
+        :avocado: tags=all,pool,pr,daily_regression,full_regression
+        :avocado: tags=small
+        :avocado: tags=pool_evict,pool_evict_bad_uuid,DAOS_5610
```
Comment (phender) [inline src/tests/ftest/pool/evict_test.py:258]:
```suggestion
        :avocado: tags=all,pr,daily_regression,full_regression
        :avocado: tags=small
        :avocado: tags=pool,evict_bad_uuid,evict,DAOS_5610
```

**Review by phender — APPROVED** [review]
It may make sense to use this in `TestPool` instead of repeating code:
```
    @property
    def identifier(self):
        """Get the pool uuid or label.
        
        Returns:
            str: pool label if using labels and one is defined; otherwise the
                pool uuid

        """
        identifier = self.uuid
        if self.use_label and self.label.value is not None:
            identifier = self.label.value
        return identifier
```

**Comment by phender** [comment]
> [src/tests/ftest/util/test_utils_pool.py:634](https://github.com/daos-stack/daos/blob/c8598984a1e04b028bea2acdc80efe59062d4ac0/src/tests/ftest/util/test_utils_pool.py#L634):
> (pylint-no-member) Instance of 'DaosPool' has no 'exclude' member

We probably should fix this and not allow the `self.control_method.value == self.USE_API` mode in `def exclude` - which should have been implemented as part of https://github.com/daos-stack/daos/commit/c53b97a0db344c7a3ad452e72b5a095ab69eb691#diff-9fe75e6cc804b2b5e4638a877f400eda7a866832bd83fada7036cf82daf6a656

**Review by dinghwah — APPROVED** [review]
Both tests passed, LGTM.

**Comment by sylviachanoiyee** [comment]
Changes all under ftests. Unit test skipped.

**Pull request merged**